s3/pkg/datastore/driver/clean.go

```diff
@@ -0,0 +1,17 @@
+package driver
+
+type Closer interface {
+	Close()
+}
+
+var closers []Closer
+
+func AddCloser(closer Closer) {
+	closers = append(closers, closer)
+}
+
+func FreeCloser() {
+	for _, c := range closers {
+		c.Close()
```

Comment:
is this idempotent ?

Reply:
it depends on the module who implements the Close method.